### PR TITLE
can: nxp_s32_canxl: use clock control APIs

### PR DESCRIPTION
**Pull request by manuargue**
Use clock control API to retrieve the module's frequency and update the boards using it to provide the source clocks.

### CODE REVIEW

File: drivers/can/can_nxp_s32_canxl.c [hunk 1]
```diff
@@ -829,6 +829,11 @@ static int can_nxp_s32_init(const struct device *dev)
 		}
 	}
 
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("Clock control device not ready");
+		return -ENODEV;
+	}
```
Comment (henrikbrixandersen) [inline drivers/can/can_nxp_s32_canxl.c:835]: Shouldn't a call to `clock_control_on()` be made here?

File: drivers/can/Kconfig.nxp_s32 [hunk 2]
```diff
@@ -1,10 +1,11 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config CAN_NXP_S32_CANXL
 	bool "NXP S32 CANXL driver"
 	default y
 	depends on DT_HAS_NXP_S32_CANXL_ENABLED
+	depends on CLOCK_CONTROL
```
Comment (henrikbrixandersen) [inline drivers/can/Kconfig.nxp_s32:8]: Should we select clock control instead?

```suggestion
	select CLOCK_CONTROL
```

File: boards/arm/s32z270dc2_r52/s32z270dc2_r52.dtsi [hunk 3]
```diff
@@ -4,7 +4,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <arm/nxp/nxp_s32z27x_r52.dtsi>
```
Comment (henrikbrixandersen) [inline boards/arm/s32z270dc2_r52/s32z270dc2_r52.dtsi:7]: Was this change intentional?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, thanks! A few comments/questions:

File: drivers/can/can_nxp_s32_canxl.c (hunk 1 above)
Comment (manuargue) [reply drivers/can/can_nxp_s32_canxl.c:835]: you're right, although the default configuration have the needed clock on at init. Let's add it for completeness

File: drivers/can/Kconfig.nxp_s32 (hunk 2 above)
Comment (manuargue) [reply drivers/can/Kconfig.nxp_s32]: agree, I was hesitant on `depends on` or `select`, but after re-reading the purpose of each, it makes more sense to use `select`.

File: boards/arm/s32z270dc2_r52/s32z270dc2_r52.dtsi (hunk 3 above)
Comment (manuargue) [reply boards/arm/s32z270dc2_r52/s32z270dc2_r52.dtsi]: It was intentional, but I will open another PR to fix it. This header is being included twice in the chain of includes. No blocking for these changes so removed.

**Comment by henrikbrixandersen** [comment]
Looks good! Can you confirm that the `tests/drivers/can/timing/` still passes tests on hardware?

**Comment by manuargue** [comment]
> Looks good! Can you confirm that the `tests/drivers/can/timing/` still passes tests on hardware?

```
*** Booting Zephyr OS build zephyr-v3.4.0-4075-g6809a0770a8c ***
Running TESTSUITE can_timing
===================================================================
testing on device can@4741b000 @ 80000000 Hz
START - test_set_timing_data_max
 PASS - test_set_timing_data_max in 0.001 seconds
===================================================================
START - test_set_timing_data_min
 PASS - test_set_timing_data_min in 0.001 seconds
===================================================================
START - test_set_timing_max
 PASS - test_set_timing_max in 0.001 seconds
===================================================================
START - test_set_timing_min
 PASS - test_set_timing_min in 0.001 seconds
===================================================================
START - test_timing
testing bitrate 20000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 250 OK, sample point error 0.0%
testing bitrate 50000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 100 OK, sample point error 0.0%
testing bitrate 125000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 40 OK, sample point error 0.0%
testing bitrate 250000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 20 OK, sample point error 0.0%
testing bitrate 500000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 10 OK, sample point error 0.0%
testing bitrate 800000, sample point 80.0% (valid): prop_seg = 7, phase_seg1 = 8, phase_seg2 = 4, prescaler = 5 OK, sample point error 0.0%
testing bitrate 1000000, sample point 75.0% (valid): prop_seg = 7, phase_seg1 = 7, phase_seg2 = 5, prescaler = 
4 OK, sample point error 0.0%
testing bitrate 125000, sample point 90.0% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 40 OK, sample point error 2.5%
testing bitrate 125000, sample point 80.0% (valid): prop_seg = 7, phase_seg1 = 8, phase_seg2 = 4, prescaler = 32 OK, sample point error 0.0%
testing bitrate 125000, sample point 100.0% (invalid): OK
testing bitrate 1000001, sample point 75.0% (invalid): OK
 PASS - test_timing in 0.138 seconds
===================================================================
START - test_timing_data
testing bitrate 500000, sample point 87.5% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 10 OK, sample point error 0.0%
testing bitrate 1000000, sample point 75.0% (valid): prop_seg = 7, phase_seg1 = 7, phase_seg2 = 5, prescaler = 
4 OK, sample point error 0.0%
testing bitrate 500000, sample point 90.0% (valid): prop_seg = 6, phase_seg1 = 7, phase_seg2 = 2, prescaler = 10 OK, sample point error 2.5%
testing bitrate 500000, sample point 80.0% (valid): prop_seg = 7, phase_seg1 = 8, phase_seg2 = 4, prescaler = 8 OK, sample point error 0.0%
testing bitrate 500000, sample point 100.0% (invalid): OK
testing bitrate 8000001, sample point 75.0% (invalid): OK
 PASS - test_timing_data in 0.068 seconds
===================================================================
TESTSUITE can_timing succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [can_timing]: pass = 6, fail = 0, skip = 0, total = 6 duration = 0.210 seconds
 - PASS - [can_timing.test_set_timing_data_max] duration = 0.001 seconds
 - PASS - [can_timing.test_set_timing_data_min] duration = 0.001 seconds
 - PASS - [can_timing.test_set_timing_max] duration = 0.001 seconds
 - PASS - [can_timing.test_set_timing_min] duration = 0.001 seconds
 - PASS - [can_timing.test_timing] duration = 0.138 seconds
 - PASS - [can_timing.test_timing_data] duration = 0.068 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
```

All good in the hood ;)

**Review by henrikbrixandersen — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]
